Assets/_GG_Ref/Entity/Unit/Enemy/Enemy.cs

```diff
@@ -17,29 +32,64 @@ public class Enemy : Unit
     private void OnCollisionEnter2D(Collision2D collision)
     {
         _collisionGameObject = collision.gameObject;
-        _collisionGameObjectCentre = _collisionGameObject.transform.position;
         switch (_collisionGameObject.tag)
         {
             case "Player":
                 TakeDamage(CurrentLifePoints);
                 break;
             case "Projectile":
                 TakeDamage(MinimalDamageTaken);
-                DropDamagePopup(MinimalDamageTaken, _collisionGameObjectCentre);
+                DropDamagePopup(MinimalDamageTaken, _collisionGameObject.transform.position);
                 break;
             case "Enemy":
                 break;
         }
     }
-    protected void BaseAwake(UnitStatsSettings settings)
+    protected void BaseAwake(EnemyStatsSettings settings)
     {
         Debug.Log($"{gameObject.name} Enemy Awake");
         _rigidbody2D = GetComponent<Rigidbody2D>();
-        _circleCollider2D = GetComponent<CircleCollider2D>();
-        //_level = InitialLevel;
+        _spriteRenderer = GetComponent<SpriteRenderer>();
+        
         var movement = new Movement(gameObject, MovementMode.Seek, settings.Speed);
         movement.SetPursuingTarget(FindObjectOfType<Player>().gameObject);
         base.BaseAwake(settings, movement);
+        _rarity = Rarity.Normal;
+        Level = MinInitialLevel;
+        RestoreLifePoints();
+    }
+    public void SetRarity(Rarity rarity)
```

Comment:
Этот метод можно разделить на несколько функций, чтобы они делали более простые вещи

```suggestion
    public void SetRarity(Rarity rarity)
    {
        _rarity = rarity;
        _spriteRenderer.material.SetFloat("_OutlineWidth", GetWidth(rarity));
        _spriteRenderer.material.SetColor("_OutlineColor", GetColor(rarity));
        var multiplier = GetMultiplier(rarity);
        if (multiplier != null)
            MaximumLifePoints.AddModifier(new StatModifier(OperationType.Multiplication, multiplier));
    }
```

А приватные методы GetWidth, GetColor, GetMultiplier реализовать через switch expression - каждый из них будет весьма коротким:

```suggestion
    private float GetWidth(Rarity rarity) => rarity switch
    {
        Rarity.Normal => 0,
        Rarity.Magic => 0.02f,
        ...
        _ => throw new ArgumentOutOfRangeException(nameof(rarity), rarity, null);
    }
```